src/Tooltip/Tooltip.tsx

```diff
@@ -107,26 +133,28 @@ export default class Tooltip extends React.Component<Props> {
     const child = React.Children.only(children);
 
     return (
-      <OverlayTrigger
-        delayShow={delayMs}
-        delayHide={delayHideMs}
-        overlay={tooltip}
-        placement={placement}
-        rootClose
-        trigger={hide ? [] : ["focus", clickTrigger ? "click" : "hover"]}
-      >
-        {React.cloneElement(child, {
-          tabIndex: 0,
-          "aria-describedby": this.id,
-          className: cssClass.FOCUSABLE_TRIGGER,
-          ...(clickTrigger && {
-            onMouseDown: (e: React.SyntheticEvent) => {
-              e.preventDefault();
-            },
-          }),
-          ...child.props,
-        })}
-      </OverlayTrigger>
+      <RootCloseWrapper onRootClose={handleHideTooltip}>
+        <>
+          {React.cloneElement(child, {
+            ref: this.tooltipTarget,
+            tabIndex: 0,
+            "aria-describedby": this.id,
+            onFocus: handleShowTooltip,
+            onBlur: handleHideTooltip,
+            onMouseEnter: handleMouseEnter,
+            onMouseLeave: handleMouseLeave,
+            onMouseDown: !clickTrigger || hide ? undefined : handleOnClick,
```

Comment:
nit:  Can we move the ternary operation on the right (`hide ? undefined : handleOnClick`) to a variable or handle this logic from within `handleOnClick`?

Reply:
Also might be worth including a comment explaining this functionality